src/engraving/dom/tuplet.cpp

```diff
@@ -194,7 +194,9 @@ bool Tuplet::calcHasBracket(const DurationElement* cr1, const DurationElement* c
     }
     bool tupletStartsBeam = beamStart->elements().front() == c1;
     bool tupletEndsBeam = beamEnd->elements().back() == c2;
-    if (tupletStartsBeam && tupletEndsBeam) {
+    bool headSide = isUp() != (c1->up() || c2->up());
+    bool isCross = c1->staffMove() || c2->staffMove();
```

Comment:
Maybe it would be fixed by changing this to `c1->vStaffIdx() != c2->vStaffIdx()`. That way, all these cases would be handled the same:
<img width="365" alt="Scherm­afbeelding 2023-09-26 om 01 13 28" src="https://github.com/musescore/MuseScore/assets/48658420/24924c92-99e0-4110-888c-47bfea059218">